polaris/run/serial.py

```diff
@@ -508,50 +515,19 @@ def _run_step_as_subprocess(test_case, step, new_log_file):
     """
     logger = test_case.logger
     cwd = os.getcwd()
-    test_name = step.path.replace('/', '_')
+    logger_name = step.path.replace('/', '_')
     if new_log_file:
         log_filename = f'{cwd}/{step.name}.log'
-        step.log_filename = log_filename
         step_logger = None
     else:
         step_logger = logger
         log_filename = None
-    with LoggingContext(name=test_name, logger=step_logger,
+
+    step.log_filename = log_filename
```

Comment:
I don't think this change is what we want. This will set `step.log_filename = None` if `new_log_file == False`, but what we actually want to have happen is that the step keeps the log file from the test case.  I think the outer function has probably already set this but we could explicitly set `step.log_filename = test_case.log_filename`.  It could also be that it doesn't matter: that nothing is actually done with `step.log_filename` so as long as the logger is correct, we're good.

Reply:
I do think this matters because we use `step.log_filename == None` sometimes to know whether we're logging to a file or not.  For example:
https://github.com/E3SM-Project/polaris/blob/867068995958a3c2b4960deb62fb6e8c4b51241e/polaris/mesh/spherical.py#L120

Reply:
Sorry, after we looked at this together, it's right.  Thanks for your patience with me arguing with my past self...